pkgdb/src/resolver/lockfile.cc

```diff
@@ -425,6 +427,141 @@ LockfileRaw::clear()
 
 
 /* -------------------------------------------------------------------------- */
+void
+LockfileRaw::load_from_content( const nlohmann::json & jfrom )
+{
+  unsigned version = jfrom["lockfile-version"];
+  debugLog( nix::fmt( "lockfile version %d", version ) );
+
+  switch ( version )
+    {
+      case 0: *this = jfrom; break;
+      case 1: this->from_v1_content( jfrom ); break;
+      default:
+        throw InvalidLockfileException( "unsupported lockfile version",
+                                        "only v0 and v1 are supprted" );
+    }
+}
+
+static void
+lockedPackageFromCatalogDescriptor( const nlohmann::json & jfrom,
+                                    LockedPackageRaw &     pkg )
+{
+  std::string attrPath = jfrom["attr_path"];
+  std::string system   = jfrom["system"];
+
+  // This would be more appropriately moved to `evalCacheCursorForInput` where
+  // we are introducing the concept of a flake, but that code won't know where
+  // the attrPath is coming from to make that detemination.
+  pkg.attrPath = splitAttrPath( "legacyPackages." + system + "." + attrPath );
+  pkg.priority = jfrom["priority"];
+  pkg.info     = jfrom;
+  pkg.input    = LockedInputRaw();
+
+  pkg.input.url   = jfrom["locked_url"];
+  pkg.input.attrs = nix::fetchers::Attrs();
+  // These attributes are needed by the current builder, and not included in the
+  // descriptor This will not always be true, but also may not be required to
+  // build depending on the path taken for future environment builds.
```

Comment:
```suggestion
  // These attributes are needed by the current builder, and not included in the
  // descriptor. This will not always be true, but also may not be required to
  // build depending on the path taken for future environment builds.
```